### ISSUE TITLE
push-to-git.sh should support deletion of files

### ISSUE THREAD

**Issue by pmonks** [symphonyoss/contrib-toolbox#9]
Currently, it appears that the `push-to-git.sh` script doesn't remove existing files from the checked out working copy, prior to copying the latest files in.  Specifically, I expected to see a `rm -rf *` between [lines 78 & 79](https://github.com/symphonyoss/contrib-toolbox/blob/master/scripts/push-to-git.sh#L78-L79) of the script.

This is important in the case where a file that was previously in the filesystem has since been removed, including the case where a file is renamed.  While extraneous files won't necessarily cause problems (especially in documentation projects), there are cases where they *will* cause issues and therefore this script should properly mirror all changes to the incoming file set, including deletes.

**Comment by maoo** [comment]
Fixed it removing the entire checked out folder.

**Comment by maoo** [comment]
Reopening due to comment on https://github.com/symphonyoss/contrib-toolbox/commit/2737808de07e101d52d645ffe41a6233e9952076#commitcomment-23461386

The script is not intended - right now - to manage deletions, but only additions/changes; do you have a specific use case in mind where deletions are needed?

**Comment by pmonks** [comment]
Yes.  Generated documentation will often have files that change name or are removed between versions - when a namespace is renamed, for example, or when an API is renamed or relocated from one namespace to another.

By not supporting forward-propagation of deletes, this script will cause clutter to accumulate in the target repository; clutter which cannot easily be manually corrected.  Over time such clutter can easily grow to outnumber valid content.
